src/field.cpp

```diff
@@ -881,6 +881,35 @@ bool map::process_fields_in_submap( submap *const current_submap,
                                 bool destroyed = fuel->burn( frd, can_spread);
 
                                 if( destroyed ) {
+									//Check what the item is made of and spawn appropriate byproducts
```

Comment:
Please move the new code into a separate function. I don't want to scroll horizontal while reading code.